### PR TITLE
Setup spring project closes #14

### LINKED ISSUE #14: Setup base spring application
As a software developer I would like to setup a basic spring application so other developers have a starting point to build out backend functionality.

**Acceptance critera**  
- spring application setup with test endpoint
- project structure setup
- appropriate build tool configured

### PR DESCRIPTION
**Pull request by jordansimsmith**
GitHub Issue (If applicable): #14 

## PR Type

What kind of change does this PR introduce?
- Feature
- Build or CI related changes

## What is the current behavior?

There is no project at all.

## What is the new behavior?

- Java Spring boot application setup.
- Used maven as a build tool (probably more used in the group over gradle)
- Test controller with test endpoint (`GET /hello`) setup as a model for future developers
- Controller unit tests setup

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jordansimsmith** [comment]
> Hey @jordansimsmith `mvn test` is working for me, are we also able to run the server locally yet? If so, how?

Yes, you can run `mvn package` and then run the jar in the target folder if you want to run from the command line. Alternatively, you can just run the project with intellij (green button)

**Comment by HarrisonLeach1** [comment]
> > Hey @jordansimsmith `mvn test` is working for me, are we also able to run the server locally yet? If so, how?
> 
> Yes, you can run `mvn package` and then run the jar in the target folder if you want to run from the command line. Alternatively, you can just run the project with intellij (green button)

Cool works for me now,  just had incorrect version on JAVA_HOME.

File: server/src/test/java/com/feeder/server/controllers/HelloControllerTests.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.feeder.server.controllers;
+
+import com.feeder.server.models.Hello;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
```
Comment (Minus20Five) [inline server/src/test/java/com/feeder/server/controllers/HelloControllerTests.java]: Recommend add annotation
```suggestion

@SpringBootTest
```

File: server/src/test/java/com/feeder/server/controllers/HelloControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class HelloControllerTests {
+
+    private HelloController subject;
+
+    @BeforeEach
+    public void setUp() {
+        this.subject = new HelloController();
+    }
```
Comment (Minus20Five) [inline server/src/test/java/com/feeder/server/controllers/HelloControllerTests.java]: Recommend injecting
```suggestion
    @Autowired private HelloController subject;
```

**Comment by jordansimsmith** [comment]
> Two thoughts:
> 
> * Is there any reason we are using java 13 instead of LTS (11)? Just thinking we might run into some weird issues and it might be hard to debug.
> * Also I still think we should be on webflux. At the very least, webflux makes it easy to parallelize REST calls out of the box (since thats basically all we're doing). And we get some (small) performance benefits since we're mostly IO-bound.

@Minus20Five JDK downgraded to 11 (LTS), webflux dependency added in 139cf97

**Pull request merged**